lib/puppet/type/user.rb

```diff
@@ -693,9 +692,8 @@ def delimiter
       defaultto false
     end
 
-    def generate
-      if !self[:purge_ssh_keys].empty? && self[:purge_ssh_keys] != :false
-        return [] if self[:ensure] == :present && !provider.exists? 
+    def eval_generate
+      if !self[:purge_ssh_keys].empty?
```

Comment:
Ah we lost the `self[:purge_ssh_keys] != :false` that was added for PUP-11131 Let me update that.